cloud/amazon/iam.py

```diff
@@ -0,0 +1,354 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: iam
+short_description: Manage IAM users, groups, roles and keys
+description:
+     - Allows for the management of IAM users, groups, roles and access keys.
+version_added: "1.9"
+options:
+  iam_type:
+    description:
+      - Type of IAM resource
+    required: "true"
+    default: null
+    choices: [ "user", "group", "role"]
+    aliases: []
+  name:
+    description: name of IAM resource
+    required: "true"
+    aliases: []
+  new_name:
+    description: when state is update, will replace name with new_name on IAM resource
+    required: "false"
+    aliases: []
+  new_path:
+    description: when state is update, will replace the path with new_path on the IAM resource
+    required: "false"
+    aliases: []
+  state:
+    description:
+      - Whether to create, delete or update the IAM resource. Note, roles cannot be updated currently.
+    required: "true"
+    default: null
+    choices: [ "present", "absent", "update" ]
+    aliases: []
+  path:
+    description:
+      - when creating, specify the path where you want the resource to be made
+    required: "false"
+    default: "/"
+    aliases: []
+  access_key_state:
+    description:
+      - when type is user: creates, removes, deactivates or activates a user's access key. Note that remove and deactivate will apply to all keys.
+    required: "false"
+    default: null
+    choices: ["create", "remove", "activate", "deactivate"]
+    aliases: []
+  group:
+    description:
+      - when type is user, adds user to the specified group. When 'update' is specified, it will add the user to groups, preserving original group membership.
```

Comment:
The distinction between `state == present` and `state == update` is a bit unclear. Is it that `state == present` will remove unspecified group memberships for this user?